card/src/main/java/com/adyen/checkout/card/StoredCardDelegate.kt

```
@@ -88,86 +208,90 @@ class StoredCardDelegate(
+            else -> R.string.checkout_kcp_birth_date_or_tax_number_hint
+        }
     }
 
-    override fun isCvcHidden(): Boolean {
-        return cardConfiguration.isHideCvcStoredCard || noCvcBrands.contains(cardType)
-    }
+    private fun getSupportedCardTypes(): List<CardType> = emptyList()
 
-    override fun isSocialSecurityNumberRequired(): Boolean {
-        return false
-    }
+    override fun isDualBrandedFlow(cardOutputData: CardOutputData): Boolean = false
 
-    override fun isKCPAuthRequired(): Boolean {
-        return false
+    override fun isInstallmentsRequired(cardOutputData: CardOutputData): Boolean {
+        return cardOutputData.installmentOptions.isNotEmpty()
     }
 
-    override fun requiresInput(): Boolean {
-        return !cardConfiguration.isHideCvcStoredCard
-    }
+    private fun mapComponentState(
+        encryptedCard: EncryptedCard,
+        stateOutputData: CardOutputData,
+        cardNumber: String,
+        firstCardType: CardType?,
+    ): CardComponentState {
+        val cardPaymentMethod = CardPaymentMethod().apply {
+            type = CardPaymentMethod.PAYMENT_METHOD_TYPE
 
-    override fun isHolderNameRequired(): Boolean {
-        return false
-    }
+            storedPaymentMethodId = getPaymentMethodId()
 
-    override fun getAddressFormUIState(
-        addressConfiguration: AddressConfiguration?,
-        addressVisibility: AddressVisibility
-    ): AddressFormUIState {
-        return AddressFormUIState.NONE
-    }
+            if (!isCvcHidden()) {
+                encryptedSecurityCode = encryptedCard.encryptedSecurityCode
+            }
 
-    override fun isAddressRequired(addressFormUIState: AddressFormUIState): Boolean {
-        return false
-    }
+            try {
+                threeDS2SdkVersion = ThreeDS2Service.INSTANCE.sdkVersion
+            } catch (e: ClassNotFoundException) {
+                Logger.e(TAG, "threeDS2SdkVersion not set because 3DS2 SDK is not present in project.")
```

We're swallowing the error here - wouldn't it be better to crash?

---

Adding a comment 👍